dll/win32/browseui/shellbrowser.cpp

```diff
@@ -786,11 +787,8 @@ HRESULT CShellBrowser::Initialize()
 
     fToolbarProxy.Initialize(m_hWnd, clientBar);
 
-
-    // create status bar
-    fStatusBar = CreateWindow(STATUSCLASSNAMEW, NULL, WS_CHILD | WS_VISIBLE | WS_CLIPSIBLINGS |
-                    SBT_NOBORDERS | SBT_TOOLTIPS, 0, 0, 500, 20, m_hWnd, (HMENU)0xa001,
-                    _AtlBaseModule.GetModuleInstance(), 0);
+    if (fStatusBarVisible)
+        CreateStatusBar();
```

Comment:
Looking at the Serge's review comments I'm thinking now, maybe keep the status bar window creation unconditional, but then immediately hide it, if it's not enabled? This way it would not be needed to introduce new function.

Reply:
I would vote for keeping it like proposed by the author for 3 reasons:
1.) the current state has been tested already
2.) I don't think it is good design to create something and then immediately hide it. Might also be less efficient from a performance / and or rendering perspective
3.) The function is not exported via the header even, so I think it is the appropriate solution chosen: If you want to call a piece of code at two positions, have a function for it to not repeat yourself.

Avoiding 2 calls would be good, sure. If that would be possible somehow: Fine.
But create+hide seems too hacky to me.

Reply:
That could make sense, and instead of "immediately hiding it", you could remove that `WS_VISIBLE` style passed in the window creation, so that the status bar is initially hidden. Then you call the ShowWindow suitably.

Reply:
^ just wanted to say this :)

Reply:
Reduced the diff in 0a3b5d04 using the flags, tested - it works. 🙂

Reply:
I admit: It is the better solution, since we had a flag for for visibility within aleady anyway. Solves also the quetion about the new funcs ignored/superfluous retval.